tor/onionfile/onion_file.go

```diff
@@ -0,0 +1,111 @@
+package onionfile
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"os"
+
+	"github.com/lightningnetwork/lnd/keychain"
+	"github.com/lightningnetwork/lnd/lnencrypt"
+	"github.com/lightningnetwork/lnd/tor"
+)
+
+var (
+	// ErrEncryptedTorPrivateKey is thrown when a tor private key is
+	// encrypted, but the user requested an unencrypted key.
+	ErrEncryptedTorPrivateKey = errors.New("it appears the Tor private key " +
+		"is encrypted but you didn't pass the --tor.encryptkey flag. " +
+		"Please restart lnd with the --tor.encryptkey flag or delete " +
+		"the Tor key file for regeneration")
+)
+
+// OnionFile is a file-based implementation of the OnionStore interface that
+// stores an onion service's private key.
+type OnionFile struct {
```

Comment:
Should this instead be called `EncryptedFile` and the existing `OnionFile` be moved to this package as well for consistency? Otherwise just from the name it's not really clear what the difference between `tor.OnionFile` and `onionfile.OnionFile` is.